### PR TITLE
maint: refactor config and simplify main()

### PR DESCRIPTION
**Pull request by robbkidd**
## Which problem is this PR solving?

- Followup to the refactoring in #111, this intends to make the lambda extension easier to understand and maintain.

## Short description of the changes

- Add a new `extension.Config` struct and `extension.NewConfigFromEnvironment()`
  function to encapsulate the reading of config env vars and setting default values.
  Add tests for this. It lives in the `extension` package--previously only a home for
  our client to the AWS Extensions API--because this is the config for our extension,
  right?

- eventpublisher grows a little from being only a thin wrapper around a libhoney
  client instance to also being responsible for reading libhoney batch send responses
  when debug is enabled. Decreases the complexity of `main()`.

- Reorder `main()` so that the actions that are taken only when running in the
  Lambda runtime environment (e.g. registering the extension, subscribing to logs)
  occur after the localMode-only blocking wait for context to be canceled. One less
  `if localMode` branch in `main()`.

- logsapi/server: Rename the start function to `StartLogsReceiver` to make it more
  clear what role the logsapi server plays in the extension.

- logsapi/client: refactor `Subscribe()` to encapsulate the creation of a client for
  the AWS Logs API and subscribing to configured log types. Simplifies `main()`.

- Fix OS signal handling. Add os.Interrupt to handle CTRL-C (generally used in
  localMode). Reorder shutdown message to before context is cancelled so it has
  time to appear on the console before process exit.

### CODE REVIEW

File: extension/config.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	// batch send to complete in this amount of time.
+	defaultBatchSendTimeout = time.Second * 15
+
+	// It's very generous to expect an HTTP connection to
+	// to be established in this time.
+	defaultConnectTimeout = time.Second * 3
+)
+
+type Config struct {
+	ApiKey                         string // Honeycomb API key
```
Comment (MikeGoldsmith) [inline extension/config.go:26]: Nicer to use same name as old config struct - `APIKey`

**Review by MikeGoldsmith — APPROVED** [review]
Looks great, good work @robbkidd - No more shall we have `config config.Config` 😁 

Left one minor suggestion for config field names.

File: extension/config.go (hunk 1 above)
Comment (robbkidd) [reply extension/config.go:26]: How much nicer is keeping the name if the previous Config struct never shipped? 

I think it boils down to which naming style would we like from here on out.

Comment (robbkidd) [reply extension/config.go:26]: For further comparison, this [replaces](https://github.com/honeycombio/honeycomb-lambda-extension/pull/112/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261L44-L46) an `apiKey` variable that used to be set in `main`.

Comment (robbkidd) [reply extension/config.go:26]: Well, [we're all over the place](https://github.com/search?l=Go&p=3&q=org%3Ahoneycombio+apikey&type=Code) as to a convention for "Api" vs "API". Let's go with your suggestion for "API" here. 👍🏻

**Comment by robbkidd** [comment]
Confirmed these changes behave in a manual integration test.

**Pull request merged**